core/src/main/java/org/elasticsearch/common/util/BloomFilter.java

```diff
@@ -321,7 +319,7 @@ static long optimalNumOfBits(long n, double p) {
         long bitCount;
 
         BitArray(long bits) {
-            this(new long[Ints.checkedCast(LongMath.divide(bits, 64, RoundingMode.CEILING))]);
+            this(new long[Ints.checkedCast(1 + bits / 64)]);
```

Comment:
You are absolutely right. That was silly of me and a good catch by you. I addressed this in ad256cc1fcaf7dcdda0f94cd161878bbdd6dbd79.